src/manifest.ts

```diff
@@ -463,26 +476,123 @@ export class Manifest {
+    }
+    const packages = await this.getPackagesToRelease(
+      // use the lastMergedPR.sha as a Commit: lastMergedPR.files will inform
+      // getPackagesToRelease() what packages had changes (i.e. at least one
+      // file under their path changed in the lastMergedPR such as
+      // "packages/mypkg/package.json"). These are exactly the packages we want
+      // to create releases/tags for.
+      [{sha: lastMergedPR.sha, message: '', files: lastMergedPR.files}],
+      lastMergedPR.sha
+    );
+    const releases: Record<string, GitHubReleaseResponse | undefined> = {};
+    let allReleasesCreated = !!packages.length;
+    for (const pkg of packages) {
+      const {releaseType, draft, ...options} = pkg.releaserOptions;
+      const releaserClass = factory.releasePRClass(releaseType);
+      const releasePR = new releaserClass({github: this.gh, ...options});
+      const pkgName = (await releasePR.getPackageName()).name;
+      const pkgLogDisp = `${releaserClass.name}(${pkgName})`;
+      if (!pkg.lastVersion) {
+        // a user manually modified the manifest file on the release branch
+        // right before merging it and deleted the entry for this pkg.
+        this.checkpoint(
+          `Unable to find last version for ${pkgLogDisp}.`,
+          CheckpointType.Failure
+        );
+        releases[pkg.path] = undefined;
+        continue;
+      }
+      this.checkpoint(
+        'Creating release for ' + `${pkgLogDisp}@${pkg.lastVersion}`,
+        CheckpointType.Success
+      );
+      const releaser = new GitHubRelease({
+        github: this.gh,
+        releasePR,
+        draft,
+      });
+      let release: ReleaseCreateResponse | undefined;
+      try {
+        release = await releaser.createRelease(pkg.lastVersion, lastMergedPR);
+      } catch (err) {
+        // There is no transactional bulk create releases API. Previous runs
+        // may have failed due to transient infrastructure problems part way
+        // through creating releases. Here we skip any releases that were
+        // already successfully created.
+        //
+        // Note about `draft` releases: The GitHub API Release unique key is
+        // `tag_name`. However, if `draft` is true, no git tag is created. Thus
+        // multiple `draft` releases can be created with the exact same inputs.
+        // (It's a tad confusing because `tag_name` still comes back populated
+        // in these calls but the tag doesn't actually exist).
+        // A draft release can even be created with a `tag_name` referring to an
+        // existing tag referenced by another release.
+        // However, GitHub will prevent "publishing" any draft release that
+        // would cause a duplicate tag to be created. release-please manifest
+        // users specifying the "release-draft" option could run into this
+        // duplicate releases scenario. It's easy enough to just delete the
+        // duplicate draft entries in the UI (or API).
```

Comment:
I spent more time just re-writing this comment. And as I did so, it occurred to me that maybe I should be checking for existing releases prior to creating, rather than relying on the API to `"already_exists"`. That would avoid this potential "duplicate draft releases" issue all together. However, it's kind of an edge case and not hard to manually clean up from. thoughts?

Reply:
I'd start with just adding a `TODO` comment, that suggests the eventual improvement.